src/Idephix/Extension/Slack/Slack.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace Idephix\Extension\Slack;
+
+use Idephix\IdephixInterface;
+use Idephix\Extension\IdephixAwareInterface;
+
+/**
+ * Description of Slack wrapper
+ *
+ * @author dymissy
+ */
+class Slack implements IdephixAwareInterface
+{
+    /**
+     * @var \Idephix\IdephixInterface
+     */
+    private $idx;
+
+    private $settings;
+
+    public function __construct($args = array())
+    {
+        $defaults = array(
+            'url' => '',
+            'channel' => '#general',
+            'icon_url' => 'https://slack.com/img/icons/app-57.png',
+            'username' => 'slackbot'
+        );
+
+        $this->settings = array_merge($defaults, $args);
+    }
+
+    public function sendToSlack($message, $attachments = array(), $channel = '', $icon_url = '', $username = '')
+    {
+        if (!$channel) {
+            $channel = $this->settings['channel'];
+        }
+
+        if (!$icon_url) {
+            $icon_url = $this->settings['icon_url'];
+        }
+
+        if (!$username) {
+            $username = $this->settings['username'];
+        }
+
+        $data = "payload=" . json_encode(array(
+                "channel" => $channel,
+                "text" => $message,
+                "icon_url" => $icon_url,
+                "unfurl_links" => true,
+                "username" => $username,
+                "attachments" => $attachments
+            ));
+
+        $result = $this->request($data, $this->settings['url']);
```

Comment:
What do you think to change the name of ```request()```method ? It isn't a verb and imo methods should be verb.

Do you like a name as ```send``` or ```doRequest``` ? For example Guzzle use ```send``` method.

In terms of readability

```$response = $this->send($data, $url)```

is better than

```$result = $this->request($data, $this->settings['url']);````